### PR TITLE
release: add v0.3.0-rc.1 release notes

### PR DESCRIPTION
**Pull request by Xunzhuo**
Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

File: release-notes/v0.3.0-rc1.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      Added APIs to Manage Envoy Deployment 
+      Added Support for Request Authentication 
+      Added Support for Global Rate Limiting
+      Added Support for Routes ReferenceGrant
+      Added Support for Namespace Server Config Type
+
+  - area: ci-tooling-testing
+    change: |
+      Fixes Make Image Failed in Darwin
+      Fixes Wait for Job Succeeded before conformance est
```
Comment (danehans) [inline release-notes/v0.3.0-rc1.yaml:29]: s/est/test/

**Review by danehans — COMMENTED** [review]
I found 1 typo, otherwise LGTM.

**Comment by danehans** [comment]
@Xunzhuo please add https://github.com/envoyproxy/gateway/issues/950 as a release note for GRPCRoute support.

cc: @arkodg

**Review by zirain — COMMENTED** [review]
lgtm, thanks!

**Review by danehans — DISMISSED** [review]

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-v0.2.0
+v0.3.0-rc.1
```
Comment (danehans) [inline VERSION:1]: `VERSION` should not be manually updated. It gets updated with `make docs-release TAG=v${MAJOR_VERSION}.${MINOR_VERSION}` which should only be run for releases and not release candidates.

**Review by danehans — CHANGES_REQUESTED** [review]

File: VERSION (hunk 2 above)
Comment (danehans) [reply VERSION:1]: VERSION needs to be updated so the version command reflects the proper version.

**Review by danehans — APPROVED** [review]

**Pull request merged**